src/mock_miner.cpp

```diff
@@ -287,9 +292,13 @@ VbkBlock MockMiner::applyVTBs(const BlockIndex<VbkBlock>& tip,
   if (!tree.acceptBlock(containingBlock, state)) {
     throw std::domain_error(state.toString());
   }
+  // a hack to prevent the block from being deleted
+  tree.getBlockIndex(containingBlock.getHash())->addRef(0);
+
   if (!tree.addPayloads(containingHash, vtbs, state)) {
     auto* containingIndex = tree.getBlockIndex(containingHash);
     VBK_ASSERT(containingIndex != nullptr);
+    containingIndex->removeRef(0);
```

Comment:
I think we should add private `acceptBlock` which will encapsulate this adding/removing logic for refs, then use it here